src/pdf_beamtime/src/pdf_beamtime_server.cpp

```diff
@@ -81,22 +84,63 @@ rclcpp_action::CancelResponse PdfBeamtimeServer::handle_cancel(
 void PdfBeamtimeServer::execute(
   const std::shared_ptr<rclcpp_action::ServerGoalHandle<PickPlaceControlMsg>> goal_handle)
 {
-  // A cool Moveit function
-  std::vector<double> home_angles = node_->get_parameter("home_angles").as_double_array();
+  const auto goal = goal_handle->get_goal();
+  //  Variables for feedback and results
+  auto feedback = std::make_shared<PickPlaceControlMsg::Feedback>();
+  auto results = std::make_shared<PickPlaceControlMsg::Result>();
+  bool fsm_results = true;
+  auto goal_home = node_->get_parameter("home_angles").as_double_array();
+  feedback->status = get_state_completions();
+
+  RCLCPP_INFO(node_->get_logger(), "Current state is state %d", static_cast<int>(current_state_));
+  if (current_state_ != State::HOME) {
+    fsm_results = reset_fsm(State::HOME, goal_home);
+  }
+  // Another array to sequencial-list the joint angles to be passed
```

Comment:
```suggestion
  // Another array to sequential-list the joint angles to be passed
```

Reply:
This line was removed as the code was refactored.